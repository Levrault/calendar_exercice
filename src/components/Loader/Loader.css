@keyframes spin {
  100% { 
    transform: rotate(360deg); 
  } 
}

.loader {
  align-items: center;
  bottom: 0;
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: fixed;
  width: 100%;
  z-index: $foreground;
}

.overlay {
  background-color: $palette-light-primary;
  bottom: 0;
  left: 0;
  opacity: .7;
  position: fixed;
  top: resolve($space-xl);
  width: 100%;
}

.icon {
  animation: spin 1s linear infinite;
  font-size: $space-xxl;
}
