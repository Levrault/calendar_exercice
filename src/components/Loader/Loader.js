import React from 'react';
import classnames from 'classnames';
import styles from './Loader.css';

/**
 * Loader
 * @param {bool} active
 */
const Loader = () => {
  return (
    <div className={styles.loader}>
      <div className={styles.overlay} />
      <i className={classnames('material-icons', styles.icon)}>autorenew</i>
    </div>
  );
};

export default Loader;
