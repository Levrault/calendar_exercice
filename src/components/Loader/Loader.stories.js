import React from 'react';
import { storiesOf } from '@storybook/react';
import { withMarkdownNotes } from '@storybook/addon-notes';
import Provider from '../../tools/storybook/Provider';
import Loader from './Loader';

storiesOf('Loader', module)
  .addDecorator(story => <Provider story={story()} />)
  .add('default', withMarkdownNotes(`
    # Loader

    A simple loading screen
    `)(() => (
    <Loader />
  )));

