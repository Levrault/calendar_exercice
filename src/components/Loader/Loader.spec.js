import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './Loader';

describe('Loader component', () => {
  it('should render without problem', () => {
    const component = renderer.create(
      <Loader />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
