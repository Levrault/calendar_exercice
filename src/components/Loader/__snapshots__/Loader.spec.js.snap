// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render without problem 1`] = `
<div>
  <div />
  <i
    className="material-icons"
  >
    autorenew
  </i>
</div>
`;
